sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -729,7 +736,9 @@ pi_result piQueueRelease(pi_queue Queue) {
 }
 
 pi_result piQueueFinish(pi_queue) {
-  DIE_NO_IMPLEMENTATION;
+  // TODO/FIXME : Only in-order queue is called for this API. Support
```

Comment:
`piQueueFinish` must guarantee that all previously enqueued commands had completed. The fact that the queue is in-order does not guarantee that. In-order queue just means enqueued commands will execute in the enqueue order. Still, they can can execute asynchronously with the thread which enqueues the commands.

As far as I understand your current implementation, it is safe for `piQueueFinish` to be no-op just because *all* the commands like kernel execution or memory copy are blocking - enqueue of any command with ESIMD EMU plugin does not return until the command is actually finished.

If so, please change the comment.

Reply:
Changed the comment.